### DISCUSSION TITLE
Reset tesla_auth

### DISCUSSION THREAD

**Discussion by johnnyoshika** [adriankumpf/tesla_auth#61] [Q&A] [answered]
I'm using the macOS precompiled binary. I signed in once using my Tesla email/password and was able to generate an access + refresh token. Now every time I run the program again, it gives me a new access + refresh token without the opportunity to sign in to another account. I have a couple of questions that I hope the community can help with:

1) How can I revoke my Tesla sign in so that I can either create token for another Tesla account or prevent someone who uses my computer from generating tokens with my auth?

2) I even downloaded a new copy of `tesla_auth` and it still didn't revoke my previous sign in. I even went as far as changing my Tesla password and surprisingly, it didn't revoke my auth. The newly generated access + refresh tokens still work successfully. How can I prevent this?

Thanks in advance!

**Comment by adriankumpf** [comment, marked as answer]
In macOS each application's WebViews have their own persistent cookie storage. That's why it keeps you logged in to your Tesla Account and immediately generate a new token pair. 

I am not aware of any way to programmatically purge the cookies of a WebView. But I will release a new version that will allow manual deletion of cookies via right click > `Inspect Element` > `Cookies`.

**Comment by johnnyoshika** [comment]
That's fantastic, thank you @adriankumpf!
